src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java

```diff
@@ -0,0 +1,111 @@
+package com.thealgorithms.scheduling;
+
+import java.util.PriorityQueue;
+
+/**
+ * This class implements the Non-Preemptive Priority Scheduling algorithm.
+ * Processes are executed in order of their priority. The process with the
+ * highest priority (lower priority number) is executed first,
+ * and once a process starts executing, it cannot be preempted.
+ */
+public final class NonPreemptivePriorityScheduling {
+
+    private NonPreemptivePriorityScheduling() {
+    }
+
+    static class Process implements Comparable<Process> {
+        int id; // Process ID
+        int burstTime; // Time required by the process for execution
+        int priority; // Priority of the process (lower value indicates higher priority)
```

Comment:
Shouldn't the process also have start time? Usually in scheduling they don't start all at once

Reply:
You are right, I will add the `startTime` attribute as well.